--- apps/Makefile.orig	2010-06-10 02:02:22.000000000 +0900
+++ apps/Makefile	2010-06-10 02:03:21.000000000 +0900
@@ -28,6 +28,7 @@
 DLIBSSL=../libssl.a
 LIBCRYPTO=-L.. -lcrypto
 LIBSSL=-L.. -lssl
+LIBCOMPAT=-lcompat
 
 PROGRAM= openssl
 
@@ -155,7 +156,7 @@
 	elif [ -n "$(FIPSCANLIB)" ]; then \
 	  FIPSLD_CC="$(CC)"; CC=$(TOP)/fips/fipsld; export CC FIPSLD_CC; \
 	fi; \
-	LIBRARIES="$(LIBSSL) $(LIBKRB5) $(LIBCRYPTO)" ; \
+	LIBRARIES="$(LIBSSL) $(LIBKRB5) $(LIBCRYPTO) $(LIBCOMPAT)" ; \
 	[ "x$(FIPSCANLIB)" = "xlibfips" ] && LIBRARIES="$$LIBRARIES -lfips"; \
 	$(MAKE) -f $(TOP)/Makefile.shared -e \
 		CC="$${CC}" APPNAME=$(EXE) OBJECTS="$(PROGRAM).o $(E_OBJ)" \
