diff --git a/tests/atom_explode/dotest b/tests/atom_explode/dotest
index 7941713..6140646 100755
--- a/tests/atom_explode/dotest
+++ b/tests/atom_explode/dotest
@@ -1,11 +1,15 @@
 #!/bin/bash
 
+find_printf() {
+	for x in `find . -mindepth 2 -type f`;do echo $x| sed 's/^..//'; basename $x; done
+}
+
 d=$PWD
 [[ $PORTDIR == "" ]] && PORTDIR=$(portageq envvar PORTDIR)
 cd ${PORTDIR}/metadata/cache || { echo SKIPPED; exit 0 ; }
 
 if [[ $1 == -v ]] ; then
-	find . -mindepth 2 -type f -printf '%P\n%f\n' | while read line ; do
+	find_printf | while read line ; do
 		echo "TESTING $line: "
 		"$d"/q "${line}"
 	done
@@ -13,11 +17,11 @@ if [[ $1 == -v ]] ; then
 fi
 
 echo -n "atom_explode():"
-time find . -mindepth 2 -type f -printf '%P\n%f\n' | "$d"/q > "$d"/me || exit $?
+time find_printf | "$d"/q > "$d"/me || exit $?
 echo
 
 echo -n "portage:"
-time find . -mindepth 2 -type f -printf '%P\n%f\n' | "$d"/atom-explode.py > "$d"/py || exit $?
+time find_printf | "$d"/atom-explode.py > "$d"/py || exit $?
 echo
 
 if diff -U 0 "$d"/me "$d"/py ; then
diff --git a/tests/qfile/dotest b/tests/qfile/dotest
index 0a4f652..035d0eb 100755
--- a/tests/qfile/dotest
+++ b/tests/qfile/dotest
@@ -24,7 +24,7 @@ EXPECTED_RESULTS=( \
 	[8]="/bin/bash" \
 )
 
-for i in `seq 1 ${#TEST_CASES[@]}` ; do
+for i in `perl -e "print join(\"\n\", 1..${#TEST_CASES[@]}),\"\n\";"` ; do
 	if [[ $(eval ${TEST_CASES[$i]}) != "${EXPECTED_RESULTS[$i]}" ]]; then
 		echo "Test #${i} failed: ${TEST_CASES[$i]}" > /dev/stderr
 		echo " - expected result was: ${EXPECTED_RESULTS[$i]}" > /dev/stderr
diff --git a/tests/source/dotest b/tests/source/dotest
index c43727b..0fb7199 100755
--- a/tests/source/dotest
+++ b/tests/source/dotest
@@ -100,7 +100,7 @@ testit src.style
 
 
 # Stupid BSD makes us check for this..
-if [ $(type -P md5sum) != "" ]; then
+if [ -n "$(type -P md5sum)" ]; then
 	for x in $(find ../.. '(' -name '*.c' -o -name '*.h' ')' ); do
 		python space.py $x > $x~
 		if [[ $(md5sum $x | awk '{print $1}') != $(md5sum $x~ | awk '{print $1}') ]]; then
