diff --git a/sandbox-1.6/libsandbox/libsandbox.c b/sandbox-1.6/libsandbox/libsandbox.c
index 91fa2b6..9784076 100644
--- a/sandbox-1.6/libsandbox/libsandbox.c
+++ b/sandbox-1.6/libsandbox/libsandbox.c
@@ -40,8 +40,14 @@
 #define LOG_FMT_CMDLINE			"FORMAT: C - Command Line\n"
 
 #define PROC_DIR			"/proc"
+#ifdef __DragonFly__
+#define LINPROC_DIR			"/compat/linux/proc"
+#define PROC_SELF_FD                   LINPROC_DIR "/self/fd"
+#define PROC_SELF_CMDLINE              PROC_DIR "/curproc/cmdline"
+#else
 #define PROC_SELF_FD			PROC_DIR "/self/fd"
 #define PROC_SELF_CMDLINE		PROC_DIR "/self/cmdline"
+#endif
 
 char sandbox_lib[SB_PATH_MAX];
 
