diff --git a/share/timedef/Makefile b/share/timedef/Makefile
index 7f01183..c720ceb 100644
--- a/share/timedef/Makefile
+++ b/share/timedef/Makefile
@@ -109,10 +109,14 @@ all: ${LOCALES:S/$/.out/g}
 
 afterinstall:
 	for l in ${LOCALES}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+			   ${LOCALEDIR}/$$l; \
 		${INSTALL} -m 644 -o ${BINOWN} -g ${BINGRP} $$l.out \
 			   ${LOCALEDIR}/$$l/LC_TIME; \
 	done
 	for l in ${LATIN15LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-15; \
 		${LN} -fs ../$$l.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-15/LC_TIME; \
 	done
@@ -129,95 +133,159 @@ afterinstall:
 #		       ${LOCALEDIR}/$$l.CP1131/LC_TIME; \
 #	done
 	for l in ${DE_LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-1; \
 		${LN} -fs ../de_DE.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-1/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-15; \
 		${LN} -fs ../de_DE.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-15/LC_TIME; \
 	done
 	for l in ${IT_LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-1; \
 		${LN} -fs ../it_IT.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-1/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-15; \
 		${LN} -fs ../it_IT.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-15/LC_TIME; \
 	done
 	for l in ${FR_LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-1; \
 		${LN} -fs ../fr_FR.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-1/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-15; \
 		${LN} -fs ../fr_FR.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-15/LC_TIME; \
 	done
 	for l in ${LN_LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l; \
 		${LN} -fs ../la_LN.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l/LC_TIME; \
 	done
 	for l in ${NL_LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-1; \
 		${LN} -fs ../nl_NL.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-1/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-15; \
 		${LN} -fs ../nl_NL.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-15/LC_TIME; \
 	done
 	for l in ${US_LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-1; \
 		${LN} -fs ../en_US.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-1/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-15; \
 		${LN} -fs ../en_US.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-15/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.US-ASCII; \
 		${LN} -fs ../en_US.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.US-ASCII/LC_TIME; \
 	done
 	for l in ${GB_LINKS}; do \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-1; \
 		${LN} -fs ../en_GB.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-1/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.ISO8859-15; \
 		${LN} -fs ../en_GB.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.ISO8859-15/LC_TIME; \
+		${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+		       ${LOCALEDIR}/$$l.US-ASCII; \
 		${LN} -fs ../en_GB.ISO8859-1/LC_TIME \
 		       ${LOCALEDIR}/$$l.US-ASCII/LC_TIME; \
 	done
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/en_GB.US-ASCII
 	${LN} -fs ../en_GB.ISO8859-1/LC_TIME \
 	       ${LOCALEDIR}/en_GB.US-ASCII/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/en_US.US-ASCII
 	${LN} -fs ../en_US.ISO8859-1/LC_TIME \
 	       ${LOCALEDIR}/en_US.US-ASCII/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/af_ZA.ISO8859-1
 	${LN} -fs ../en_US.ISO8859-1/LC_TIME \
 	       ${LOCALEDIR}/af_ZA.ISO8859-1/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/af_ZA.ISO8859-15
 	${LN} -fs ../en_US.ISO8859-15/LC_TIME \
 	       ${LOCALEDIR}/af_ZA.ISO8859-15/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/ko_KR.CP949
 	${LN} -fs ../ko_KR.eucKR/LC_TIME \
 	       ${LOCALEDIR}/ko_KR.CP949/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/zh_CN.GBK
 	${LN} -fs ../zh_CN.GB2312/LC_TIME \
 	       ${LOCALEDIR}/zh_CN.GBK/LC_TIME
 	for l in ${UTF8_LATIN1LINKS}; do \
+	       ${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	             ${LOCALEDIR}/$$l.UTF-8; \
 	       ${LN} -fs ../$$l.ISO8859-1/LC_TIME \
 	             ${LOCALEDIR}/$$l.UTF-8/LC_TIME; \
 	done
 	for l in ${DE_LINKS}; do \
+	       ${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	              ${LOCALEDIR}/$$l.UTF-8; \
 	       ${LN} -fs ../de_DE.UTF-8/LC_TIME \
 	              ${LOCALEDIR}/$$l.UTF-8/LC_TIME; \
 	done
 	for l in ${IT_LINKS}; do \
+	       ${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	              ${LOCALEDIR}/$$l.UTF-8; \
 	       ${LN} -fs ../it_IT.UTF-8/LC_TIME \
 	              ${LOCALEDIR}/$$l.UTF-8/LC_TIME; \
 	done
 	for l in ${FR_LINKS}; do \
+	       ${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	              ${LOCALEDIR}/$$l.UTF-8; \
 	       ${LN} -fs ../fr_FR.UTF-8/LC_TIME \
 	              ${LOCALEDIR}/$$l.UTF-8/LC_TIME; \
 	done
 	for l in ${NL_LINKS}; do \
+	       ${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	              ${LOCALEDIR}/$$l.UTF-8; \
 	       ${LN} -fs ../nl_NL.UTF-8/LC_TIME \
 	              ${LOCALEDIR}/$$l.UTF-8/LC_TIME; \
 	done
 	for l in ${US_LINKS}; do \
+	       ${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	              ${LOCALEDIR}/$$l.UTF-8; \
 	       ${LN} -fs ../en_US.UTF-8/LC_TIME \
 	              ${LOCALEDIR}/$$l.UTF-8/LC_TIME; \
 	done
 	for l in ${GB_LINKS}; do \
+	       ${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	              ${LOCALEDIR}/$$l.UTF-8; \
 	       ${LN} -fs ../en_GB.UTF-8/LC_TIME \
 	              ${LOCALEDIR}/$$l.UTF-8/LC_TIME; \
 	done
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/en_IE.UTF-8
 	${LN} -fs ../en_GB.UTF-8/LC_TIME \
 	       ${LOCALEDIR}/en_IE.UTF-8/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/af_ZA.UTF-8
 	${LN} -fs ../en_US.UTF-8/LC_TIME \
 	       ${LOCALEDIR}/af_ZA.UTF-8/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/zh_HK.UTF-8
 	${LN} -fs ../zh_TW.UTF-8/LC_TIME \
 	       ${LOCALEDIR}/zh_HK.UTF-8/LC_TIME
+	${INSTALL} -d -m 755 -o ${BINOWN} -g ${BINGRP} \
+	       ${LOCALEDIR}/zh_HK.Big5HKSCS
 	${LN} -fs ../zh_TW.UTF-8/LC_TIME \
 	       ${LOCALEDIR}/zh_HK.Big5HKSCS/LC_TIME
 
